specification/sections/02-providers.md

```diff
@@ -250,6 +253,15 @@ class MyProvider implements Provider, AutoDisposable {
   }
 ```
 
+#### Requirement 2.5.2
+
+> After a provider's shutdown function has terminated successfully, the provider's state **MUST** revert to its uninitialized state.
+
+If a provider requires initialization, once it's shut down, it should transition to its initial `NOT_READY` state, so it can be reinitialized.
```

Comment:
>  ... so it can be reinitialized

I think we should not enforce this behavior through spec 🤔

Reply:
Do you mean we should just remove the "so it can be reinitialized"?

We could remove it, but I think that this is the logical implication of transitioning back to `NOT_READY`. On the other hand, I guess you could make the argument some providers could still be unable to be made `READY` again, and they could throw if they are reinitialized.

Reply:
Yes, originally thought we shouldn't define this through the spec. But with your explanation, I think we should reword it to have **MAY** - ex:-  ... so it **MAY** be reinitialized.